src/main/java/io/cdap/plugin/gcp/dataplex/common/util/DataplexUtil.java

```diff
@@ -0,0 +1,414 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.gcp.dataplex.common.util;
+
+import com.google.api.gax.core.FixedCredentialsProvider;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.auth.oauth2.ServiceAccountCredentials;
+import com.google.cloud.dataplex.v1.DataplexServiceClient;
+import com.google.cloud.dataplex.v1.DataplexServiceSettings;
+import com.google.cloud.dataplex.v1.Job;
+import com.google.cloud.dataplex.v1.JobName;
+import com.google.cloud.dataplex.v1.MetadataServiceClient;
+import com.google.cloud.dataplex.v1.MetadataServiceSettings;
+import com.google.cloud.dataplex.v1.TaskName;
+import com.google.common.collect.Lists;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.etl.api.FailureCollector;
+import io.cdap.plugin.gcp.bigquery.util.BigQueryTypeSize;
+import io.cdap.plugin.gcp.bigquery.util.BigQueryUtil;
+import io.cdap.plugin.gcp.common.GCPConnectorConfig;
+import io.cdap.plugin.gcp.common.GCPUtils;
+
+import org.apache.hadoop.conf.Configuration;
+import org.awaitility.Awaitility;
+import org.awaitility.core.ConditionTimeoutException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nullable;
+
+/**
+ * Common Util class for dataplex source and sink plugins.
+ */
+public final class DataplexUtil {
+
+  private static final Logger LOG = LoggerFactory.getLogger(DataplexUtil.class);
+  private static final String INPUT_DIR = "mapreduce.input.fileinputformat.inputdir";
+  private static final String SERVICE_ACCOUNT_JSON = "google.cloud.auth.service.account.json";
+  private static final String SERVICE_ACCOUNT_KEYFILE = "google.cloud.auth.service.account.json.keyfile";
+
+  /**
+   * Converts Entity Schema into a CDAP Schema object.
+   *
+   * @param dataplexSchema Dataplex Schema to be converted.
+   * @param collector      Failure collector to collect failure messages for the client.
+   * @return CDAP schema object
+   */
+  public static Schema getTableSchema(com.google.cloud.dataplex.v1.Schema dataplexSchema,
```

Comment:
It is a bit confusing that the method has 'table' in it but there is no mention of a table inside the method

Reply:
dataplexSchema is also a table schema in dataplex

Reply:
renamed dataplexSchema to dataplexTableSchema